### PR TITLE
Refactoring, created a finger table service and more

### PR DESCRIPTION
**Pull request by stienis96**
Refactoring, created a finger table service that keeps track of the finger table and predecessor. Keep track of interval and such for every finger entry. And moved some methods to a Util class.

### CODE REVIEW

**Comment by stienis96** [comment]
@l8518 when running 0, 1, 3, 6 (in that order) the finger table of 6 doesn't get updated correctly for interval [2-6) the last entry. (M = 3 btw). This is also true for the develop version.

**Comment by stienis96** [comment]
Okay I fixed the bug with the last commit.

File: src/main/java/org/distributed/systems/ChordStart.java [hunk 1]
```diff
@@ -2,18 +2,19 @@
...
 import akka.actor.Props;
-import org.distributed.systems.chord.actors.Node;
+import org.distributed.systems.chord.actors.NodeActor;
 
 public class ChordStart {
 
     public static final int STANDARD_TIME_OUT = 10000;
     // FIXME 160 according to sha-1 but this is the max_length of a java long..
-    public static final int M = 64;
+    public static final int M = 3; // Number of bits in key nodeId's
```
Comment (l8518) [inline src/main/java/org/distributed/systems/ChordStart.java]: Is this also working for the later version, that we need to benchmark? I would assume M is larger in that case?

Maybe use a ENV and config var

File: src/main/java/org/distributed/systems/ChordStart.java [hunk 2]
```diff
@@ -2,18 +2,19 @@
...
+    public static final long AMOUNT_OF_KEYS = Math.round(Math.pow(2, M));
 
     public static void main(String[] args) {
         // Create actor system
         ActorSystem system = ActorSystem.create("ChordNetwork"); // Setup actor system
 
-        system.actorOf(Props.create(Node.class), "ChordActor");
+        system.actorOf(Props.create(NodeActor.class), "ChordActor");
```
Comment (l8518) [inline src/main/java/org/distributed/systems/ChordStart.java:19]: :+1: I think that really helps having a bit more concise code!

File: src/main/java/org/distributed/systems/chord/messaging/Predecessor.java [hunk 3]
```diff
@@ -12,12 +10,10 @@ public Request() {
     }
 
     public static class Reply implements Response {
-        public ActorRef predecessor;
-        public long predecessorId;
+        public ChordNode predecessor;
 
-        public Reply(ActorRef predecessor, long predecessorId) {
```
Comment (l8518) [inline src/main/java/org/distributed/systems/chord/messaging/Predecessor.java:18]: Perfect! I found my implementation quite annoying, and wanted to encapsulated that too! :+1:

File: src/main/java/org/distributed/systems/chord/util/Util.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package org.distributed.systems.chord.util;
+
+import com.typesafe.config.Config;
+import org.distributed.systems.ChordStart;
+import org.distributed.systems.chord.util.impl.HashUtil;
+
+import java.net.InetAddress;
+
+public class Util {
```
Comment (l8518) [inline src/main/java/org/distributed/systems/chord/util/Util.java:9]: :clap:

**Review by l8518 — COMMENTED** [review]
Looks good to me! Only question would regarding the value of `M` in ChordStart

File: src/main/java/org/distributed/systems/ChordStart.java (hunk 1 above)
Comment (stienis96) [reply src/main/java/org/distributed/systems/ChordStart.java]: - [x] Yeah something like env var would be nice. 👍

**Comment by l8518** [comment]
> @l8518 when running 0, 1, 3, 6 (in that order) the finger table of 6 doesn't get updated correctly for interval [2-6) the last entry. (M = 3 btw). This is also true for the develop version.

Guess I exepected some bugs, thanks for fixing!

File: src/main/java/org/distributed/systems/ChordStart.java (hunk 1 above)
Comment (l8518) [reply src/main/java/org/distributed/systems/ChordStart.java]: Okay, guess let's add that to Trello

**Review by l8518 — APPROVED** [review]
Haha never mind!

**Pull request merged**